pkg/relayer/proxy/proxy.go

```diff
@@ -0,0 +1,110 @@
+package proxy
+
+import (
+	"context"
+
+	sdkclient "github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	accounttypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+
+	blocktypes "pocket/pkg/client"
+	"pocket/pkg/observable"
+	"pocket/pkg/observable/channel"
+	"pocket/x/service/types"
+	sessiontypes "pocket/x/session/types"
+	suppliertypes "pocket/x/supplier/types"
+)
+
+var (
+	_ RelayerProxy = &relayerProxy{}
+)
+
+type relayerProxy struct {
+	// keyName is the supplier's key name in the keyring. It is used along with the keyring to
+	// get the supplier address and sign the relay responses.
+	keyName string
+	keyring keyring.Keyring
+
+	// blocksClient is the client used to get the latest block height from the blockchain
+	// and be notified about new blocks to update the current session when needed.
```

Comment:
```suggestion
	// and be notified of new incoming blocks. One of its uses is to update the current session data.
```

Reply:
It is exactly its usage in the context of the `RelayerProxy`. IMO, we should stick to `It is used to ...` in this comment and will definitely update it when its usage evolves (probably with ring signatures)

Reply:
> It and will definitely update it when its usage evolves (probably with ring signatures)

Sounds good. I won't push hard on it.

Personally, I've never people do a good job at updating documentation unless someone (e.g. the reviewer) remembers to tell the implementor to change documentation.

As long as you remember to enforce that as well, I'm okay with a 🤝